pig/src/main/java/com/netflix/iceberg/pig/PigParquetReader.java

```diff
@@ -0,0 +1,523 @@
+            throw new UnsupportedOperationException("Unsupported type: " + primitive.getOriginalType());
+        }
+      }
+
+      switch (primitive.getPrimitiveTypeName()) {
+        case FIXED_LEN_BYTE_ARRAY:
+        case BINARY:
+          return new BytesReader(desc);
+        case BOOLEAN:
+        case INT32:
+        case INT64:
+        case FLOAT:
+        case DOUBLE:
+          return new UnboxedReader<>(desc);
+        default:
+          throw new UnsupportedOperationException("Unsupported type: " + primitive);
+      }
+    }
+
+    private String[] currentPath() {
+      String[] path = new String[fieldNames.size()];
+      if (!fieldNames.isEmpty()) {
+        Iterator<String> iter = fieldNames.descendingIterator();
+        for (int i = 0; iter.hasNext(); i += 1) {
+          path[i] = iter.next();
+        }
+      }
+
+      return path;
+    }
+
+    protected String[] path(String name) {
+      String[] path = new String[fieldNames.size() + 1];
+      path[fieldNames.size()] = name;
+
+      if (!fieldNames.isEmpty()) {
+        Iterator<String> iter = fieldNames.descendingIterator();
+        for (int i = 0; iter.hasNext(); i += 1) {
+          path[i] = iter.next();
+        }
+      }
+
+      return path;
+    }
+  }
+
+  private static class StringReader extends PrimitiveReader<String> {
+    StringReader(ColumnDescriptor desc) {
+      super(desc);
+    }
+
+    @Override
+    public String read(String reuse) {
+      Binary binary = column.nextBinary();
+      ByteBuffer buffer = binary.toByteBuffer();
+      if (buffer.hasArray()) {
+        return new String(
+            buffer.array(), buffer.arrayOffset() + buffer.position(), buffer.remaining(), UTF_8);
+      } else {
+        return new String(binary.getBytes(), UTF_8);
+      }
+    }
+  }
+
+  private static class DateReader extends PrimitiveReader<String> {
+    private static final OffsetDateTime EPOCH = Instant.ofEpochSecond(0).atOffset(ZoneOffset.UTC);
+
+    DateReader(ColumnDescriptor desc) {
+      super(desc);
+    }
+
+    @Override
+    public String read(String reuse) {
+      OffsetDateTime day = EPOCH.plusDays(column.nextInteger());
+      return String.format("%04d-%02d-%02d", day.getYear(), day.getMonth().getValue(), day.getDayOfMonth());
+    }
+  }
+
+  private static class BytesReader extends PrimitiveReader<DataByteArray> {
+    BytesReader(ColumnDescriptor desc) {
+      super(desc);
+    }
+
+    @Override
+    public DataByteArray read(DataByteArray reuse) {
+      byte[] bytes = column.nextBinary().getBytes();
+      return new DataByteArray(Arrays.copyOf(bytes, bytes.length));
+    }
+  }
+
+  private static class TimestampMicrosReader extends UnboxedReader<Long> {
+    TimestampMicrosReader(ColumnDescriptor desc) {
+      super(desc);
+    }
+
+    @Override
+    public Long read(Long ignored) {
+      return column.nextLong() / 1000;
+    }
+
+  }
+
+  private static class BinaryDecimalReader extends PrimitiveReader<BigDecimal> {
```

Comment:
We should move these decimal and String readers into ParquetValueReaders because they aren't specific to Pig. It doesn't need to be done in this commit, though.